flink-runtime/src/main/java/org/apache/flink/runtime/state/ttl/TtlUpdateType.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.state.ttl;
+
+/**
+ * This option value configures when to prolong state TTL.
+ */
+public enum TtlUpdateType {
+	/** TTL is disabled. State does not expire. */
+	Disabled,
```

Comment:
Seems like this is never needed? Why would somebody register TTL state and then declare it disabled?

Reply:
This is more for internally named configuration, basically for later use in TTL wrapper factory to decide whether to wrap with TTL or not. Alternative would be probably to save null'ed, Optional config or negative TTL in state descriptor when configured w/o TTL by default. I found it more explicit, although, it faces user and I planned that he will rather probably just not configure TTL at all than use this option.

Reply:
I think we can still decide during the next step when TTL is activated in state descriptors along with builders